src/db.rs

```diff
@@ -197,10 +195,15 @@ impl Db {
         if let Some(prev) = prev {
             if let Some(when) = prev.expires_at {
                 // clear expiration
-                state.expirations.remove(&(when, key));
+                state.expirations.remove(&(when, key.clone()));
             }
         }
 
+        // Track the expiration.
+        if let Some(when) = expires_at {
+            state.expirations.insert((when, key));
+        }
```

Comment:
Please add a comment that explains why we need to insert _after_ the call to `remove`.